makefile

```diff
@@ -1,8 +1,17 @@
-VERSION ?= 5.2.0
-BRANCH ?= develop
+VERSION  := 5.2.0
+BRANCH   := develop
+ARTIFACT := prodbin-$(VERSION)-$(BRANCH).tar.gz
 
-.PHONY: all
-all: tar
+.PHONY: all clean build javascript
 
-tar:
-	tar cfz prodbin-$(VERSION)-$(BRANCH).tar.gz Products bin
\ No newline at end of file
+all: build
+
+build: javascript
+	tar cvfz $(ARTIFACT) Products bin
+
+javascript:
+	make -f javascript.mk build
+
+clean:
+	rm -f $(ARTIFACT)
+	-make -f javascript.mk clean
```

Comment:
Yes - the dash says - don't exit with an error on this step if it fails.  On Jenkins builds, we always want to execute `make clean build`, though some build slaves may not have any files to clean